docs/guides/ReferralProgramDelegationGuideForContracts.md

```diff
@@ -0,0 +1,108 @@
+
+N**o action is required if you’re using an EOA as a referral partner or trader. Only contracts need to delegate.**
+
+## How to Delegate?
+
+Your contract can delegate through our `vePERPReferralRewardDelegate` contract, and the address is `0x2dE8e18BDaef25C2DE0bED29C8B72E49261CA88d` on Optimism mainnet. You could also check the contract on [Etherscan.](https://optimistic.etherscan.io/address/0x2dE8e18BDaef25C2DE0bED29C8B72E49261CA88d) You could check out the source code on GitHub: [perpetual-protocol/voting-escrow](https://github.com/perpetual-protocol/voting-escrow/blob/main/contracts/RewardDelegate.sol).
+
+Here is the interface of `vePERPReferralRewardDelegate`:
+
+```solidity
+interface IRewardDelegate {
+    // should be called by the partner/trader contract, to delegate to an EOA who locks vePERP on the partner/trader contract's behalf
+    function setBeneficiaryCandidate(address candidate) external;
+
+    // should be called by the EOA (beneficiary), to confirm the delegation
+    function updateBeneficiary(address truster) external;
+
+    // should be called by the partner/trader contract, if you decided to revoke the delegation
+    function clearBeneficiary(address beneficiary) external;
+
+    function getBeneficiaryCandidate(address truster) external view returns (address);
+
+    function getBeneficiaryAndQualifiedMultiplier(address user) external view returns (address, uint256);
+}
+```
+
+Assume that you’re using `PartnerContract` as a referral partner, the account that owns a referral code; Also, `TraderContract` as a referral trader, the account that uses a referral code to trade. And you would like to delegate both `PartnerContract` and `TraderContract`’s referral rewards to an EOA `Alice`.
+
+First, you will need to modify your `PartnerContract` and `TraderContract` to implement `setBeneficiaryCandidate()`:
+
+```solidity
+interface IRewardDelegate {
+    function setBeneficiaryCandidate(address candidate) external;
+    function clearBeneficiary(address beneficiary) external;
+}
+
+// do the same thing for TraderContract
+contract PartnerContract is OwnableUpgradeable {
+    ...
+
+    // add the following functions to your contract
+    function setBeneficiaryCandidate(address candidate) external onlyOwner {
+        IRewardDelegate(rewardDelegate).setBeneficiaryCandidate(candidate);
+    }
+
+    function clearBeneficiary(address beneficiary) external onlyOwner {
+        IRewardDelegate(rewardDelegate).clearBeneficiary(beneficiary);
+    }
+}
```

Comment:
Should we demonstrate how to set `rewardDelegate` first ? or it's unnecessary ?

Reply:
Vinta saied we publish as-is for now